### ISSUE TITLE
Is there record mode to delete the current cassette and write new cassette?

### ISSUE THREAD

**Issue by FayeFang** [vcr/vcr#628]
There are four record mode: :all, :new_episodes, :none, :once. They all keep the old cassette.

I am using :all currently, but it keeps lots of unused and old requests on file. So what I want to do is to create a new record mode to delete current cassette and write a new one.

Any experience with this? Thanks!

**Comment by ecnalyr** [comment]
I am fairly confident there is not a method to delete unused cassettes.

One big reason may be because we don't have a way of reliably knowing if a recorded request is truly unused.

Please see [this issue](https://github.com/vcr/vcr/issues/587) that contains a similar request/suggestion and the details I wrote in the first comment.

**Comment by FayeFang** [comment]
Lol @ecnalyr the person who posted that issue is a co-worker of mine. Apparently it has been a persistent issue for us! 

I actually have a small hack to make what I want to happen. In the code below, whenever the user wants to re-record any particular cassette, the current one will be deleted first. This way, the user is not deleting all cassettes, but the only relevant cassette the user wants to re-record.

And I don't see it being any dangerous. What do you think?

```
module CassetteCleaner
  def insert_cassette(name, options = {})
    if configuration.default_cassette_options[:record] == :all
      current_cassette = VCR::Cassette.new(name, options).file
      File.delete(current_cassette) if File.exist?(current_cassette)
    end

    super
  end
end
VCR.extend(CassetteCleaner)
```

**Comment by ecnalyr** [comment]
I think it would be reasonable to put it up for review if it required both

- a flag to be enabled 
and 
- is clearly documented that it is destructive to any previously recorded requests.

I have to make it clear that I have the breadth of knowledge of VCR that other contributors/maintainers have and there's a chance there's a good reason why something like this doesn't already exist.

But I like the idea of having a clean PR up for review.

**Comment by FayeFang** [comment]
Sure I can put up a pr for review.

**Comment by mcfiredrill** [comment]
There is also `re_record_interval`:
https://relishapp.com/vcr/vcr/v/3-0-3/docs/configuration/default-cassette-options#cassettes-can-set-their-own-options

**Comment by krainboltgreene** [comment]
Honestly having dealt with this I think we just need to document how to be
more surgical with VCR.

**Comment by ryanseys** [comment]
@ecnalyr 

What about if this was implemented as a new `overwrite` record mode? It would behave exactly like the `all` record mode, except it would delete the current cassette file if it exists beforehand?

I'm happy to try to throw together a PR if this sounds reasonable?

**Comment by ryanseys** [comment]
I also see there is this PR that needs to be back-merged? https://github.com/vcr/vcr/pull/769 

If I work on rebasing that, would it be a reasonable way to solve this problem?

**Comment by ryanseys** [comment]
I rebased into a new PR an option that should allow this functionality out of the box: https://github.com/vcr/vcr/pull/946

Please take a look!

**Comment by ryanseys** [comment]
https://github.com/vcr/vcr/pull/946 is now shipped. Not yet released, but we can mark this as fixed.

**Comment by olleolleolle** [comment]
Fixed in #946.
